gb-joypad/src/config.rs

```diff
@@ -0,0 +1,81 @@
+use crate::InputType;
+use sdl2::keyboard::Scancode;
+use serde::{
+    de::{Error, MapAccess, Visitor},
+    ser::SerializeMap,
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::collections::HashMap;
+use std::fmt;
+use std::marker::PhantomData;
+
+/// Store a joypad configuration.
+///
+/// Since it implement Serialise and Deserialize, it can be used to quickly save/load a joypad configuration into/from a file.
+pub struct Config {
+    mapping: HashMap<Scancode, InputType>,
+}
+
+impl Config {
+    pub fn from_mapping(mapping: HashMap<Scancode, InputType>) -> Self {
+        Config { mapping }
+    }
+
+    pub fn mapping(self) -> HashMap<Scancode, InputType> {
+        self.mapping
+    }
+}
+
+impl Serialize for Config {
```

Comment:
To use `#[derive(Serialize)]` all its components have to implement `Serialize`but the sdl2 `Scancode` does not, and we can't implement it directly since this type is not from our crate.

Reply:
valid point